From 40e3cc07eb48043f96d29f7393099de6ba8f3407 Mon Sep 17 00:00:00 2001
From: Nicolas Vuaille <nicolas.vuaille@kitware.com>
Date: Thu, 23 Jan 2025 10:49:58 +0100
Subject: [PATCH] Remove reference to old Python pygments formatters module

We used to have a pygments fork that adds a custom formatters,
namely `redhtmltab` [1]. This is fork is no more in ThirdParty [2]
and not used in Superbuild [3] (it uses upstream pypi).

Also, this seems to adress the crash described at #22849

[1] https://gitlab.kitware.com/third-party/pygments
[2] see 649ca4e52da58c6918b8a5140d71b170f7639dbe
[3] https://gitlab.kitware.com/paraview/common-superbuild/-/blob/master/versions.cmake
---
 Qt/Python/pqPythonSyntaxHighlighter.cxx | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/Qt/Python/pqPythonSyntaxHighlighter.cxx b/Qt/Python/pqPythonSyntaxHighlighter.cxx
index 5412fd756b2..a9ac18cbae8 100644
--- a/Qt/Python/pqPythonSyntaxHighlighter.cxx
+++ b/Qt/Python/pqPythonSyntaxHighlighter.cxx
@@ -44,18 +44,10 @@ pqPythonSyntaxHighlighter::pqPythonSyntaxHighlighter(QObject* p, QTextEdit& text
       this->HighlightFunction.TakeReference(
         PyObject_GetAttrString(this->PygmentsModule, "highlight"));
       vtkSmartPyObject lexersModule(PyImport_ImportModule("pygments.lexers"));
-      vtkSmartPyObject formattersModule(PyImport_ImportModule("pygments.formatters.redtabhtml"));
+      vtkSmartPyObject formattersModule(PyImport_ImportModule("pygments.formatters"));
       vtkSmartPyObject htmlFormatterClass;
-      // If we have the custom formatter written for ParaView, great.
-      // otherwise just default to the HtmlFormatter in pygments
       if (formattersModule)
       {
-        htmlFormatterClass.TakeReference(
-          PyObject_GetAttrString(formattersModule, "RedTabHtmlFormatter"));
-      }
-      else
-      {
-        formattersModule.TakeReference(PyImport_ImportModule("pygments.formatters"));
         htmlFormatterClass.TakeReference(PyObject_GetAttrString(formattersModule, "HtmlFormatter"));
       }
 
-- 
GitLab

